### PR TITLE
Update Changelog

### PR DESCRIPTION
**Pull request by neha-bhargava**

### CODE REVIEW

File: changelog.txt [hunk 1]
```diff
@@ -1,3 +1,17 @@
+4.24.0
+============
+
+New Features:
+**Performance improvement by reducing the number of calls to GetAllAccessTokens()**. See issue [#2204](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/issues/2204) for details.
+**MSAL .NET now sends the whether the token cache is serialized to telemetry**. See issue [#2185](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/issues/2185) for details.
```
Comment (jennyf19) [inline changelog.txt:6]: might re-read this...not sure what you're trying to say ;)

File: changelog.txt [hunk 2]
```diff
@@ -1,3 +1,17 @@
+4.24.0
+============
+
+New Features:
+**Performance improvement by reducing the number of calls to GetAllAccessTokens()**. See issue [#2204](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/issues/2204) for details.
```
Comment (jennyf19) [inline changelog.txt:5]: I think this would be a bug fix, but idk.

File: changelog.txt [hunk 3]
```diff
@@ -1,3 +1,17 @@
...
+**MSAL .NET now sends the whether the token cache is serialized to telemetry**. See issue [#2185](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/issues/2185) for details.
+
+Bug Fixes:
+**MSAL .NET now do not force user to enter credentials for WAM AcquireTokenInteractive with picker**. See issue [#2233](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/issues/2233) for details.
```
Comment (jennyf19) [inline changelog.txt:9]:
```suggestion
**MSAL .NET will not force the user to enter their credentials when logging-in with WAM**. See issue [#2233](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/issues/2233) for details. 
```

File: changelog.txt [hunk 4]
```diff
@@ -1,3 +1,17 @@
+4.24.0
+============
+
+New Features:
+**Performance improvement by reducing the number of calls to GetAllAccessTokens()**. See issue [#2204](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/issues/2204) for details.
+**MSAL .NET now sends the whether the token cache is serialized to telemetry**. See issue [#2185](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/issues/2185) for details.
+
+Bug Fixes:
+**MSAL .NET now do not force user to enter credentials for WAM AcquireTokenInteractive with picker**. See issue [#2233](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/issues/2233) for details. 
+**MSAL .NET now throws `RopcDoesNotSupportMsaAccounts` when ROPC is attempted with MSA account**. See issue [#2169](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/issues/2169) for details.
```
Comment (jennyf19) [inline changelog.txt:10]:
```suggestion
**MSAL .NET now throws an actionable error message when ROPC is attempted with MSA accounts**. See issue [#2169](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/issues/2169) for details.
```

File: changelog.txt [hunk 5]
```diff
@@ -1,3 +1,17 @@
...
+**MSAL .NET now supports WithForceRefresh for AcquireTokenOnBehalfOfParameterBuilder**. See issue [#2232](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/issues/2232) for details.
```
Comment (jennyf19) [inline changelog.txt:11]:
```suggestion
**MSAL .NET now supports `WithForceRefresh` as part of the `AcquireTokenOnBehalfOfParameterBuilder`**. See issue [#2232](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/issues/2232) for details.
```

File: changelog.txt [hunk 6]
```diff
@@ -1,3 +1,17 @@
...
+**Fix PlatformNotSupportedException in MacOS**. See issue [#2251](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/issues/2251) for details.
+**AcquireTokenOnBehalfOfAsync is now not visible in the Intellisense**. See issue [#2263](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/issues/2263) for details.
```
Comment (jennyf19) [inline changelog.txt:13]:
```suggestion
```

Comment (jennyf19) [reply changelog.txt:13]: i don't think this is needed in the changelog

File: changelog.txt [hunk 7]
```diff
@@ -1,3 +1,17 @@
...
+**MSAL .NET now sends the whether the token cache is serialized to telemetry**. See issue [#2185](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/issues/2185) for details.
+
+Bug Fixes:
+**MSAL .NET now do not force user to enter credentials for WAM AcquireTokenInteractive with picker**. See issue [#2233](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/issues/2233) for details. 
+**MSAL .NET now throws `RopcDoesNotSupportMsaAccounts` when ROPC is attempted with MSA account**. See issue [#2169](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/issues/2169) for details.
+**MSAL .NET now supports WithForceRefresh for AcquireTokenOnBehalfOfParameterBuilder**. See issue [#2232](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/issues/2232) for details.
+**Fix PlatformNotSupportedException in MacOS**. See issue [#2251](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/issues/2251) for details.
```
Comment (jennyf19) [inline changelog.txt:12]:
```suggestion
**Fix `PlatformNotSupportedException` in MacOS due to custom cache not being defined as part of the build script**. See issue [#2251](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/issues/2251) for details.
```

Comment (jennyf19) [reply changelog.txt:12]: @bgavrilMS to confirm this one ^

File: changelog.txt (hunk 2 above)
Comment (pmaytak) [inline changelog.txt:5]:
```suggestion
**`AcquireTokenForClient` and `AcquireTokenSilent` have improved performance, especially for large token caches**. See issue [#2204](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/issues/2204) for details.
```

**Review by pmaytak — APPROVED** [review]

**Comment by pmaytak** [comment]
@jennyf19 Do you think it would be helpful to change this `changelog.txt` to `.md`, same as in Identity Web?

File: changelog.txt (hunk 2 above)
Comment (jmprieur) [reply changelog.txt:5]: yes, or fundamental work. It's not a new feature.

File: changelog.txt (hunk 1 above)
Comment (jmprieur) [inline changelog.txt:6]:
```suggestion
**MSAL .NET now sends telemetry when token serialization happens**. See issue [#2185](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/issues/2185) for details.
```

File: changelog.txt (hunk 5 above)
Comment (jmprieur) [reply changelog.txt:11]: yes, or **MSAL .NET now supports `WithForceRefresh` when using `AcquireTokenOnBehalfOf` ? (the builder is an implementation detail?)

File: changelog.txt (hunk 7 above)
Comment (jmprieur) [reply changelog.txt:12]: Or just Fix `PlatformNotSupportedException` in MacOS ?

File: changelog.txt (hunk 6 above)
Comment (jmprieur) [reply changelog.txt:13]: or just cleaner IntelliSense? but yes. this is not needed

**Review by jmprieur — COMMENTED** [review]
Thanks @neha-bhargava @jennyf19 @pmaytak 
I left a few more suggestions

File: changelog.txt (hunk 1 above)
Comment (bgavrilMS) [reply changelog.txt:6]: I don't think we should advertise this. It's not a feature from a customer perspective.

File: changelog.txt [hunk 8]
```diff
@@ -1,3 +1,17 @@
+4.24.0
+============
+
+New Features:
+**`AcquireTokenForClient` and `AcquireTokenSilent` have improved performance, especially for large token caches**. See issue [#2204](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/issues/2204) for details.
```
Comment (bgavrilMS) [inline changelog.txt:5]:
```suggestion
**`AcquireTokenForClient` and `AcquireTokenSilent` have improved performance, especially for large token caches**. See issue [#2204](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/issues/2204) for details.
** TokenCache notifications are not fired when the TokenCache is not serialized by developers, improving performance of all APIs utilising the token cache ** 
```

File: changelog.txt (hunk 1 above)
Comment (jennyf19) [reply changelog.txt:6]: we have a new section, which is "Experimental Features" it was @henrik-me 's idea for the last release. i forgot about it. that would go in that section.

Comment (neha-bhargava) [reply changelog.txt:6]: This is not specific to regional. This is for all the requests. But agree with @bgavrilMS this is for internal tracking. Can remove it from here.

**Comment by jennyf19** [comment]
> @jennyf19 Do you think it would be helpful to change this `changelog.txt` to `.md`, same as in Identity Web?

@pmaytak we changed identity web based on a customer suggestion, and we only had a few releases out. MSAL has a much larger history. i think we are fine w/it as is.

**Pull request merged**